js/src/kb-search/wp-seo-kb-search.js

```diff
@@ -0,0 +1,305 @@
+import React from 'react';
+import AlgoliaSearch from 'algoliasearch';
+
+class AlgoliaSearcher extends React.Component {
+
+	/**
+	 * ALgoliaSearcher constructor.
+	 *
+	 * @constructor
+	 * @param {object} props Properties of the ALgoliaSearcher component.
+	 */
+	constructor( props ) {
+		super();
+		this.state = {
+			searchString: '',	// The by the user given search input
+			usedQueries: {}, 	// An object containing the read articles by the user per used search string.
+			results: [],		// The found results.
+			errorMessage: '',	// An description of the error if one occurs while executing the search.
+			showDetail: false,	//  Shows the search results if this is set to false. Otherwhise, it shows the content of the article which index (of the state.results array) correlates with the value of showDetail.
+			searching: false
+		};
+		this.props = props;
+		this.initAlgoliaClient();
+
+        this.searchButtonClicked = this.searchButtonClicked.bind( this );
+        this.hideDetail = this.hideDetail.bind( this );
+	}
+
+	/**
+	 * Initializes the algolia client and index variables.
+	 */
+	initAlgoliaClient() {
+        this.client = AlgoliaSearch( this.props.algoliaApplicationId, this.props.algoliaApiKey );
+        this.index = this.client.initIndex( this.props.algoliaIndexName );
+	}
+
+	/**
+	 * Handles the input changed event. It saved the used search string and performs a search.
+	 *
+	 * @param {object} e The event.
+	 */
+	searchButtonClicked( e ) {
```

Comment:
Can you rename `e` to something like  `evt` to make it a but more clear it is an event.